packages/api-client/src/views/Request/ResponseSection/ResponseBodyVirtual.vue

```diff
@@ -1,19 +1,37 @@
 <script setup lang="ts">
 import ViewLayoutCollapse from '@/components/ViewLayout/ViewLayoutCollapse.vue'
+import { useResponseBody } from '@/hooks/useResponseBody'
 import { ScalarVirtualText } from '@scalar/components'
 import { formatJsonOrYamlString } from '@scalar/oas-utils/helpers'
 import { computed } from 'vue'
 
+import ResponseBodyDownload from './ResponseBodyDownload.vue'
+
 const props = defineProps<{
   content: string
+  data: unknown
+  headers: { name: string; value: string; required: boolean }[]
 }>()
 
 const textContent = computed(() => formatJsonOrYamlString(props.content))
+
+const { mimeType, attachmentFilename, dataUrl } = useResponseBody({
+  data: props.data,
+  headers: props.headers,
+})
 </script>
 
 <template>
   <ViewLayoutCollapse class="!max-h-100% overflow-x-auto response-body-virtual">
     <template #title>Body</template>
+    <template
+      v-if="textContent && dataUrl"
+      #actions>
+      <ResponseBodyDownload
+        :filename="attachmentFilename"
+        :href="dataUrl"
+        :type="mimeType.essence" />
+    </template>
```

Comment:
does it have to be textContent to be downloaded? Can’t we download any content?

Reply:
this just check that we have the content, but doesn't affect the nature of the downloaded file

Reply:
but text content is used and pass to the virtual text component

Reply:
I guess the virtual response body never gets triggered if it's a binary blob? 🤔

Reply:
so we can just remove the `textContent` from the `v-if`, right?

Reply:
as we already have that guard here i guess we could
<img width="348" alt="image" src="https://github.com/user-attachments/assets/faf4e62a-1dcf-4dc4-86d0-4e75da6cff37" />